bigtable-hbase-parent/bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableAdmin.java

```diff
@@ -819,93 +836,172 @@ public boolean isTableAvailable(TableName tableName, byte[][] splitKeys) throws
   @Override
   public void snapshot(String snapshotName, TableName tableName)
       throws IOException, SnapshotCreationException, IllegalArgumentException {
-    throw new UnsupportedOperationException("snapshot");  // TODO
+    com.google.longrunning.Operation operation = snapshotTable(snapshotName, tableName);
+    try {
+      waitForOperation(operation.getName());
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new IOException("Create snapshot operation was interrupted", e);
+    }
+  }
+
+  private com.google.longrunning.Operation snapshotTable(String snapshotName, TableName tableName)
+      throws IOException {
+    BigtableClusterName clusterName = this.connection.getSession().getClusterName();
+    BigtableInstanceName instanceName = options.getInstanceName();
+    BigtableTableName btTableName = instanceName.toTableName(tableName.getNameAsString());
+    return bigtableTableAdminClient.snapshotTable(SnapshotTableRequest.newBuilder()
+        .setCluster(clusterName.getClusterName())
+        .setSnapshotId(snapshotName)
+        .setName(btTableName.toString())
+        .build());
+  }
+
+  /**
+   * Waits for an operation like cluster resizing to complete.
```

Comment:
It's a good point about shared code, but I think I'd like to defer the refactoring to another PR.